src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphSubmissionDAO.scala

```diff
@@ -61,12 +61,15 @@ class GraphSubmissionDAO extends SubmissionDAO with GraphDAO {
   /** create a submission (and its workflows) */
   override def save(workspaceNamespace: String, workspaceName: String, submission: Submission, txn: RawlsTransaction) =
     txn withGraph { db =>
-      val workspace = getWorkspaceVertex(db, workspaceNamespace, workspaceName).getOrElse(throw new IllegalArgumentException(s"workspace ${workspaceNamespace}/${workspaceName} does not exist"))
+      val workspaceVertex = getWorkspaceVertex(db, workspaceNamespace, workspaceName).getOrElse(throw new IllegalArgumentException(s"workspace ${workspaceNamespace}/${workspaceName} does not exist"))
       val submissionVertex = setVertexProperties(submission, addVertex(db, null))
-      workspace.addEdge(ExecutionEdgeTypes.submissionEdgeType, submissionVertex)
+      workspaceVertex.addEdge(ExecutionEdgeTypes.submissionEdgeType, submissionVertex)
       submission.workflow.foreach { workflow =>
         val workflowVertex = setVertexProperties(workflow, addVertex(db, null))
-        workspace.addEdge(ExecutionEdgeTypes.workflowEdgeType, workflowVertex)
+        val entityVertex =  getSinglePipelineResult(new GremlinPipeline(workspaceVertex).out(submission.entityType).filter(hasProperty("_name", workflow.entityName))).
```

Comment:
sorry that I missed this before: the submission entity type is not necessarily the same as the workflow entity type. A submission's entity type will usually be a set and the workflow's entity type will be a member of that set. The workflow case class needs to include an entity type.